src/main/java/com/shipmentEvents/demo/Main.java

```diff
@@ -8,11 +8,19 @@ public static void main(String[] args) {
         Scanner in = new Scanner(System.in);
 
 
-
         //adding a line here testing things
         System.out.println("Please enter your name");
         String name = in.nextLine();
 
+        do{
+            //this could be an infinite loop!
+            //String comparison using ==
+            if(name == "Morgan") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.